core/src/main/java/com/amplifyframework/core/model/ModelPrimaryKey.java

```diff
@@ -0,0 +1,184 @@
+    private static final long serialVersionUID = 1L;
+    private final Serializable key;
+    private final List<? extends Serializable> sortedKeys;
+
+    /**
+     * Constructor for Model Primary key class. Takes in partition key and an array of sort keys as parameters.
+     * @param key Partition key.
+     * @param sortedKeys Array of sort keys.
+     */
+    public ModelPrimaryKey(Serializable key, Serializable... sortedKeys) {
+        this.key = key;
+        this.sortedKeys = Arrays.asList(sortedKeys);
+    }
+
+    /**
+     * Returns the Partition key of the model.
+     * @return Partition key.
+     */
+    public Serializable key() {
+        return key;
+    }
+
+    /**
+     * Returns the Array of sort keys of the model.
+     * @return Array of sort keys.
+     */
+    public List<? extends Serializable> sortedKeys() {
+        return sortedKeys;
+    }
+
+    /**
+     * Returns a concatenated string of unique identifier. Concatenates PartitionKey#Sortkey1#sortkey2...
+     * @return string representation of unique identifier
+     */
+    public String getIdentifier() {
+        return Helper.getIdentifier(key, sortedKeys);
+    }
+
+    /**
+     * Helper class for functions related to primary key.
+     */
+    public static class Helper {
+        /**
+         * Character which is used to encapsulate a primary key field while concatenating primary key fields.
+         */
+        public static final String PRIMARY_KEY_ENCAPSULATE_CHAR = "\"";
+
+        /**
+         * Character which is used as delimiter while concatenating primary key fields.
+         */
+        public static final String PRIMARY_KEY_DELIMITER = "#";
+
+        /**
+         * Helper function which creates a query predicate which returns.
+         * @param model Model to create the predicate from.
+         * @param tableName Table name for which predicate needs to be created.
+         * @param primaryKeyList List of primary key field value list.
+         * @return Query Predicate to query with a unique identifier.
+         */
+        public static QueryPredicate getQueryPredicate(Model model, String tableName, List<String> primaryKeyList) {
+            QueryPredicate matchId = null;
+            if (!(model.resolveIdentifier() instanceof ModelPrimaryKey)) {
+                matchId = QueryField.field(tableName, primaryKeyList.get(0)).eq(model.resolveIdentifier());
+            } else {
+                ModelPrimaryKey<?> primaryKey = (ModelPrimaryKey<?>) model.resolveIdentifier();
+                Iterator<?> sortKeyIterator = primaryKey.sortedKeys().listIterator();
+                for (String key : primaryKeyList) {
+                    if (matchId == null) {
+                        matchId = QueryField.field(tableName, key).eq(primaryKey.key().toString());
+                    } else {
+                        matchId.and(QueryField.field(tableName, key).eq(sortKeyIterator.next()));
+                    }
+                }
+            }
```

Comment:
This implementation generates incorrect query predicate for `SerializeModel` code path, where if the model is with composite PK, it generate predicate against the partition key field.

This causes failure on reconciliation of newly created model, as the modelExists check always return false due to incorrect PK query field is used...


```suggestion
            Serializable resolvedIdentifier = model.resolveIdentifier();
            QueryPredicate matchId = null;
            if (!(resolvedIdentifier instanceof ModelPrimaryKey)) {
                // single attribute primary key e.g. "ID"
                if (primaryKeyList.size() == 1) {
                    matchId = QueryField.field(tableName, primaryKeyList.get(0)).eq(resolvedIdentifier);
                }
                // composite primary key for the use case of SerializedModel
                else {
                    // it looks like that needs to move `SQLiteTable.PRIMARY_KEY_FIELD_NAME` into core
                    matchId = QueryField.field(tableName, "@@primaryKey").eq(resolvedIdentifier);
                }
            } else {
                ModelPrimaryKey<?> primaryKey = (ModelPrimaryKey<?>) resolvedIdentifier;
                Iterator<?> sortKeyIterator = primaryKey.sortedKeys().listIterator();
                for (String key : primaryKeyList) {
                    if (matchId == null) {
                        matchId = QueryField.field(tableName, key).eq(primaryKey.key().toString());
                    } else {
                        matchId.and(QueryField.field(tableName, key).eq(sortKeyIterator.next()));
                    }
                }
            }
```

Reply:
It loops through primary key field and includes partition key and sort keys in the predicate.

Reply:
Removed and code updated to fix the flutter issue.